formulas/dnsmasq/files/protect-dns-resolver-config

```diff
@@ -0,0 +1,19 @@
+eval "$(
+    funcBody=$(declare -f make_resolv_conf)
+    echo "${funcBody/make_resolv_conf/make_resolv_conf_real}"
+);"
+
+make_resolv_conf() {
+    resolvConf=$(readlink -f "/etc/resolv.conf" 2>/dev/null) || resolvConf="/etc/resolv.conf"
+    if ! cp "$resolvConf" "${resolvConf}.tmp"; then
+        return 1
+    fi
+    makeResolvConfOutput=$(make_resolve_conf_real 2>&1)
```

Comment:
I would like a comment here or maybe before the `if` explaining why we capture the output and consider any output to be a failure.  It would be good to pose alternates to this solution and why those alternates fall short, considering how much time you put into this.